Source/Core/Gen4/Generators/WildGenerator4.cpp

```diff
@@ -244,6 +244,160 @@ std::vector<WildState4> WildGenerator4::generateMethodJ(u32 seed, const Encounte
     return states;
 }
 
+template <bool bug>
+void createPokemon(PokeRNG &go, WildState4 &state, u8 buffer, u8 synchNature, u8 genderRatio, u16 tsv, const Lead &lead,
+                   const StateFilter &filter)
+{
+    u32 pid = 0;
+    if constexpr (bug)
+    {
+        bool cuteCharmSuccess = false;
+        if (lead == Lead::CuteCharmFemale || lead == Lead::CuteCharm25M || lead == Lead::CuteCharm50M || lead == Lead::CuteCharm75M
+            || lead == Lead::CuteCharm875M)
+        {
+            if ((go.nextUShort<true>() % 3) != 0) // Successfull cute charm
+            {
+                // Get hunt nature
+                state.setNature(go.nextUShort<true>() % 25);
+
+                if (!filter.compareNature(state))
+                {
+                    return;
+                }
+
+                cuteCharmSuccess = true;
+                pid = buffer + state.getNature();
+            }
+        }
+
+        for (u8 loop = 0; loop < 4; loop++)
+        {
+            switch (lead)
+            {
+            case Lead::Synchronize:
+                if ((go.nextUShort<true>() & 1) == 0) // Successful synch
+                {
+                    state.setNature(synchNature);
+                }
+                else // Failed synch
+                {
+                    state.setNature(go.nextUShort<true>() % 25);
+                }
+                break;
+            case Lead::CompoundEyes:
+                state.setLead(Lead::CompoundEyes);
+                break;
+            default:
+                break;
+            }
+
+            if (!cuteCharmSuccess)
+            {
+                // Get hunt nature
+                state.setNature(go.nextUShort<true>() % 25);
+
+                // Begin search for valid pid
+                do
+                {
+                    u16 low = go.nextUShort<true>();
+                    u16 high = go.nextUShort<true>();
+                    pid = static_cast<u32>((high << 16) | low);
+                } while (pid % 25 != state.getNature());
+            }
+
+            state.setPID(pid);
+            state.setAbility(pid & 1);
+            state.setGender(pid & 255, genderRatio);
+            state.setShiny<8>(tsv, (pid >> 16) ^ (pid & 0xffff));
+
+            u16 iv1 = go.nextUShort<true>();
+            u16 iv2 = go.nextUShort<true>();
+
+            state.setIVs(iv1, iv2);
+            state.calculateHiddenPower();
+
+            if constexpr (bug)
```

Comment:
This if statement isn't necessary. We are already in a "if constexpr (bug)" section